tests/test_parallel.py

```diff
@@ -115,54 +115,49 @@ def mock_submit(fn="", *args, **kwargs):
         factory = civis.parallel.make_backend_template_factory(
             from_template_id=from_template_id,
             max_submit_retries=max_submit_retries,
-            client=create_client_mock(),
+            client=mock_client,
         )
     else:
         factory = civis.parallel.make_backend_factory(
-            max_submit_retries=max_submit_retries, client=create_client_mock()
+            max_submit_retries=max_submit_retries, client=mock_client
         )
     register_parallel_backend("civis", factory)
-    with parallel_backend("civis"):
-        # NB: joblib >v0.11 relies on callbacks from the result object to
-        # decide when it's done consuming inputs. We've mocked the result
-        # object here, so Parallel must be called either with n_jobs=1 or
-        # pre_dispatch='all' to consume the inputs all at once.
-        parallel = Parallel(n_jobs=1, pre_dispatch="n_jobs")
+    with parallel_config("civis"):
+        # NB: Since joblib v1.3.0,
+        # n_jobs=1 would just run a for loop and wouldn't use any parallel backend.
+        parallel = Parallel(n_jobs=2, pre_dispatch="n_jobs")
         if should_fail:
             with pytest.raises(civis.parallel.JobSubmissionError):
                 parallel(delayed(sqrt)(i**2) for i in range(3))
+            if from_template_id:
+                assert (
+                    mock_custom_exec_cls.return_value.submit.call_count
+                    == submit_call_count
+                )
+                mock_custom_exec_cls.return_value.submit.assert_called_with(
+                    JOBLIB_FUNC_FILE_ID=12345
+                )
+            else:
+                assert (
+                    mock_executor_cls.return_value.submit.call_count
+                    == submit_call_count
+                )
         else:
-            parallel(delayed(sqrt)(i**2) for i in range(3))
-
-
-@mock.patch.object(civis.parallel, "CustomScriptExecutor")
-@mock.patch.object(civis.parallel, "_CivisBackendResult")
-@mock.patch.object(civis.parallel.civis.io, "file_to_civis", autospec=True)
-def test_template_submit(mock_file, mock_result, mock_pool):
```

Comment:
Dropping this test as I've added the same assertion calls in `test_retries` above.